setup.py

```diff
@@ -59,6 +65,42 @@
     "license": "Apache License 2.0",
 }
 
+package_name = "scipy"
+
+scipy_package = importlib.util.find_spec(package_name)
+package_directory = path.dirname(scipy_package.origin)
+
+if system_platform == "Linux":
+    file_path_within_package_linux = "../scipy.libs/"
+    scipy_lib_path = path.join(package_directory, file_path_within_package_linux)
+    file_prefix = "libopenblasp"
+    file_extension = ".so"
+    search_pattern = f"{file_prefix}*{file_extension}"
+    openblas_so_file = glob.glob(f"{search_pattern}", root_dir=scipy_lib_path)[0]
+    openblas_lib_name = openblas_so_file[3:-3]
+    custom_calls_extension = Extension(
+        "catalyst.utils.custom_calls",
+        sources=["frontend/catalyst/utils/custom_calls.cpp"],
+        libraries=[openblas_lib_name],
+        library_dirs=[scipy_lib_path],
+    )
+
+elif system_platform == "Darwin":
+    file_path_within_package_linux = "/.dylibs/"
+    scipy_lib_path = path.join(package_directory, file_path_within_package_linux)
+    file_prefix = "libopenblas"
+    file_extension = ".dylib"
+    search_pattern = f"{file_prefix}*{file_extension}"
+    openblas_dylib_file = glob.glob(f"{search_pattern}", root_dir=scipy_lib_path)[0]
+    openblas_lib_name = openblas_dylib_file[3:-6]
+    custom_calls_extension = Extension(
+        "catalyst.utils.custom_calls",
+        sources=["frontend/catalyst/utils/custom_calls.cpp"],
+        libraries=[openblas_lib_name],
+        library_dirs=[scipy_lib_path],
+    )
```

Comment:
Do you want to add `else` and raise an error? :thinking:

Reply:
I would say no because it is written in the docs that we do not support windows